CMakeLists.txt

```diff
@@ -379,7 +379,7 @@ endif()
 function(add_warning_flags name)
   target_compile_options(${name} PRIVATE
     $<$<CXX_COMPILER_ID:MSVC>:/Zi /W4 /WX /wd4127 /wd4324 /wd4201>
-    $<$<NOT:$<OR:$<CXX_COMPILER_ID:MSVC>,$<STREQUAL:${CMAKE_CXX_SIMULATE_ID},MSVC>>>:-fno-exceptions -fno-rtti -Wall -Wextra -Werror -Wundef>
```

Comment:
To make this work, I have had to enable exceptions.  This seems less than ideal as it pulls in more dependencies and usage of libstdc++.

This could be addressed with additional build configurations.

@davidchisnall Do you have opinions on this?